source/whats-new.txt

```diff
@@ -45,6 +43,82 @@ What's New in 1.12
      deprecated. You can create a client and connect in one call
      by using the ``mongo.Connect()`` method.
 
+New features of the 1.12 Go driver release include: 
+
+Queryable Encryption
+~~~~~~~~~~~~~~~~~~~~
+
+The release adds production-ready support for
+:ref:`qe-manual-feature-qe` (QE). To use
+the QE feature, you must use MongoDB Server version 7.0 or later and
+``libmongocrypt`` version 1.8.0 or later.
+  
+This release introduces the ``ClientEncryption.CreateEncryptedCollection()``
+method, which automatically creates data encryption keys when
+creating a new encrypted collection. This method adds the ability to
+fetch KMS credentials automatically from Azure, GCP, and AWS
+environments.
+
+Logging Interface
+~~~~~~~~~~~~~~~~~
+
+You can now record detailed information about connection management and
+command execution.
+
+The logging configuration adds a new ``LogSink`` interface that is
+compatible with the `logr.LogSink <https://pkg.go.dev/github.com/go-logr/logr#LogSink>`__ interface.
+You can use existing ``logr`` adapters with the logging interface
+such as :github:`zerologr <go-logr/zerologr>` or :github:`zapr <go-logr/zapr>`.
+    
+You can also implement your own ``LogSink`` adapter. The following
+example shows how to integrate logging with an existing :github:`zerolog
+<rs/zerolog>` logger:
+
+.. code-block:: go
+
+   sink := zerologr.New(&myLogger).GetSink()
+   loggerOptions := options.Logger().SetSink(sink)
+   options.Client().SetLoggerOptions(loggerOptions)
+
+Additional BSON Marshalling Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This release adds new features to the ``options`` package to specify
+various BSON marshalling and unmarshalling behaviors.
+
+The following example shows how to set new BSON options on your
+``Client``. The options specify the following behaviors:
+
+- The driver falls back to ``json`` struct tags if ``bson`` struct tags
+  are missing.
+- The driver marshals ``nil`` Go map types as empty BSON documents.
+- The driver marshals ``nil`` Go slice types as empty BSON arrays.
+
+.. code-block:: go
+
+   bsonOpts := &options.BSONOptions{
+   	UseJSONStructTags: true,
+   	NilMapAsEmpty:     true,
+   	NilSliceAsEmpty:   true,
+   }
+   options.Client().SetBSONOptions(bsonOpts)
```

Comment:
Suggestion:
I think the example might not have enough detail, but also may not belong in What's New. I think it would be more clear to present a sample struct and show how the options impact that rather than describing it, perhaps on the "[Work with BSON](https://www.mongodb.com/docs/drivers/go/current/fundamentals/bson/#struct-tags)" page. I would probably leave as is for now, but create a ticket to defer to a more detailed example on that page.

Reply:
There is now a [full example](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#example-Connect-BSONOptions) of using `BSONOptions` in the `pkg.go.dev` docs site. Maybe we can link to that?

Reply:
I'll link to the full example for now from the What's New, but create a ticket to add detail to the Work with BSON page.